node/heartbeat/storage/heartbeatStorer.go

```diff
@@ -0,0 +1,167 @@
+package storage
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/ElrondNetwork/elrond-go/core/logger"
+	"github.com/ElrondNetwork/elrond-go/marshal"
+	"github.com/ElrondNetwork/elrond-go/node/heartbeat"
+	"github.com/ElrondNetwork/elrond-go/storage"
+)
+
+var log = logger.DefaultLogger()
+
+const peersKeysDbEntry = "keys"
+const genesisTimeDbEntry = "genesisTime"
+
+// HeartbeatDbStorer is the struct which will handle storage operations for heartbeat
+type HeartbeatDbStorer struct {
+	storer      storage.Storer
+	marshalizer marshal.Marshalizer
+}
+
+// NewHeartbeatDbStorer will create an instance of HeartbeatDbStorer
+func NewHeartbeatDbStorer(
+	storer storage.Storer,
+	marshalizer marshal.Marshalizer,
+) (*HeartbeatDbStorer, error) {
+	if storer == nil || storer.IsInterfaceNil() {
+		return nil, heartbeat.ErrNilMonitorDb
+	}
+	if marshalizer == nil || marshalizer.IsInterfaceNil() {
+		return nil, heartbeat.ErrNilMarshalizer
+	}
+
+	return &HeartbeatDbStorer{
+		storer:      storer,
+		marshalizer: marshalizer,
+	}, nil
+}
+
+// LoadGenesisTime will return the genesis time saved in the storer
+func (hs *HeartbeatDbStorer) LoadGenesisTime() (time.Time, error) {
+	genesisTimeFromDbBytes, err := hs.storer.Get([]byte(genesisTimeDbEntry))
+	if err != nil {
+		return time.Time{}, heartbeat.ErrFetchGenesisTimeFromDb
+	}
+
+	var genesisTimeFromDb time.Time
+	err = hs.marshalizer.Unmarshal(&genesisTimeFromDb, genesisTimeFromDbBytes)
+	if err != nil {
+		return time.Time{}, heartbeat.ErrUnmarshalGenesisTime
+	}
+
+	return genesisTimeFromDb, nil
+}
+
+// UpdateGenesisTime will update the saved genesis time and will log if the genesis time changed
+func (hs *HeartbeatDbStorer) UpdateGenesisTime(genesisTime time.Time) error {
+	if hs.storer.Has([]byte(genesisTimeDbEntry)) == nil { // if found, check for changes
+		genesisTimeFromDb, err := hs.LoadGenesisTime()
+		if err != nil {
+			return err
+		}
+
+		if genesisTimeFromDb != genesisTime {
+			log.Info(fmt.Sprintf("updated heartbeat's genesis time to %s", genesisTimeFromDb))
```

Comment:
this print should be done **after** the successful saving of the genesis time. Otherwise will mislead the person who will debug this code.
Also, the print should be called after the line 74. This will generate duplicate code. Extract in a func.

Reply:
Moved after save. 

What duplicate code?

Reply:
Looks, ok